packages/amplify-e2e-tests/src/__tests__/graphql-v2/invalid-input-arguments.test.ts

```diff
@@ -0,0 +1,151 @@
+import {
+    initJSProjectWithProfile,
+    deleteProject,
+    amplifyPush,
+} from 'amplify-category-api-e2e-core';
+import { addApiWithBlankSchemaAndConflictDetection, updateApiSchema, getProjectMeta } from 'amplify-category-api-e2e-core';
+import { createNewProjectDir, deleteProjectDir } from 'amplify-category-api-e2e-core';
+import { API } from 'aws-amplify';
+import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
+
+(global as any).fetch = require('node-fetch');
+
+const projectName = 'invalidarguments';
+describe('Invalid arguments should throw an error', () => {
+    let projRoot: string;
+    
+    beforeAll(async () => {
+        projRoot = await createNewProjectDir(projectName);
+        await initJSProjectWithProfile(projRoot, {
+            name: projectName,
+        });
+        const v2Schema = 'model_with_datatypes.graphql';
+        await addApiWithBlankSchemaAndConflictDetection(projRoot, { transformerVersion: 2 });
+        await updateApiSchema(projRoot, projectName, v2Schema);
+        await amplifyPush(projRoot);
+        await configureAmplifyAPI(projRoot);
+    });
+
+    afterAll(async () => {
+        await deleteProject(projRoot);
+        deleteProjectDir(projRoot);
+    });
+
+    it('create mutation should error on invalid input arguments', async () => {
+        const invalid_create_inputs = [
+            { title: 'task 1', description: 'task 1 description', priority: 1, dueDate: 'invalid date' }, // Invalid date argument
+            { title: 'test', description: 'task 1 description', priority: 2, dueDate: 1000000 }, // Invalid date argument
+            { title: 'task 1', description: 'task 1 description', priority: 5.1, dueDate: '2022-01-01T00:00:00.000Z' }, // Passing float in place of int
+            { title: 'task 1', description: 'task 1 description', priority: "invalid", dueDate: '2022-01-01T00:00:00.000Z' }, // Passing string in place of int
+        ];
+        
+        await Promise.all(
+            invalid_create_inputs.map(async (input) => {
+                try {
+                    const createResult = await createTask(input);
+                    expect(false).toBeTruthy();
```

Comment:
Does this become more readable as an `expect(async () => await createTask(input).toThrow(/matcher/)

I'm fine as-is, so treat as a question.

Reply:
I've considered this but we will lose the ability to verify error.errors is defined and error length is greater than 1.